### PR TITLE
Reduce the audio lag

### PR DESCRIPTION
**Pull request by rsn8887**
Without this change, the audio lag on Switch platform is almost
one full second, much larger than on other platforms.

This commit reduces the audio buffer which reduces the audio lag
without affecting sound quality. It reflects the audio buffer
setting used by almost all other homebrew software on the Switch.

### CODE REVIEW

File: src/platform/sound_device.c [hunk 1]
```diff
@@ -12,7 +12,12 @@
 #define AUDIO_RATE 22050
 #define AUDIO_FORMAT AUDIO_S16
 #define AUDIO_CHANNELS 2
+
+#if defined(__SWITCH__)
+#define AUDIO_BUFFERS 512
```
Comment (devnoname120) [inline src/platform/sound_device.c]: This leads to the question: should we apply this change to all the platforms?

Comment (rsn8887) [reply src/platform/sound_device.c]: it shouldn’t hurt unless it leads to crackling or stuttering.

Comment (rsn8887) [reply src/platform/sound_device.c]: On Vita: The audio lag is much less than on the Switch to begin with. If I apply this change, and click on the icons very fast, I get audio drop-outs. 

So, I think it is best to make this Switch-only for now.

Comment (devnoname120) [reply src/platform/sound_device.c]: I'd suggest adding a comment to explain why it's set to 512 for the Switch.

Comment (bvschaik) [reply src/platform/sound_device.c]: For Mac, 4096 is also too large, 512 gives a way better experience, so I would like to lower it for all platforms. How does 1024 sound on Vita? Maybe that's a suitable default.

Comment (rsn8887) [reply src/platform/sound_device.c]: 1024 seems to work great even on Vita! I updated this PR.

**Pull request merged**